src/components/MagicCodeInput.js

```diff
@@ -103,16 +109,32 @@ function MagicCodeInput(props) {
         },
     }));
 
-    useEffect(() => {
-        // Blurs the input and removes focus from the last input and, if it should submit
-        // on complete, it will call the onFulfill callback.
+    /**
+     * Check if current magic code is ready for submit
+     * @returns {boolean}
+     */
+    const isReadyToSubmit = () => {
         const numbers = decomposeString(props.value);
-        if (!props.shouldSubmitOnComplete || _.filter(numbers, (n) => ValidationUtils.isNumeric(n)).length !== CONST.MAGIC_CODE_LENGTH) {
+        return props.shouldSubmitOnComplete && _.filter(numbers, (n) => ValidationUtils.isNumeric(n)).length === CONST.MAGIC_CODE_LENGTH && !props.network.isOffline;
+    };
```

Comment:
We don't this method  logic could be moved into `validateAndSubmit` method

Reply:
@Santhosh-Sellavel Thanks, I removed it.